### ISSUE TITLE
Option to suppress messages from lsp--info

### ISSUE THREAD

**Issue by dchneric** [emacs-lsp/lsp-mode#1884]
This is more of an enhancement for Emacs lsp usability per se. We'd like to suppress the message sent by lsp--info, for example, in lsp-mode.el (around line 8187):

`(lsp--info "Connected to %s."
                     (apply 'concat (--map (format "[%s]" (lsp--workspace-print it))
                                           lsp--buffer-workspaces)))))`

The messages are displayed in emacs echo area and it shadows existing messages or prompts, for example, when we invoke lsp after opening files. Is it possible to configure lsp as "silent" mode and suppress messages coming from lsp--info or even lsp--warn?

**Comment by yyoncho** [comment]
We can and it is easy to do so. The concern is the granularity - because it might suppress useful messages as well. We may introduce "level". 0 means log everything, 10 log nothing.

**Comment by dchneric** [comment]
Thanks it would be great to have this feature!

**Comment by krei-del** [comment]
I agree, too. Messages that are not shown with lsp--info could still be part of lsp-log ...

**Comment by magnars** [comment]
If you, like me, stumbled on this issue a few years later, here's an intermediate fix for your emacs config:

```el
(defun my/silence-some-lsp-info-messages (orig-fn &rest args)
  (unless (string-equal (car args) "Connected to %s.")
    (apply orig-fun args)))

(advice-add 'lsp--info :around #'my/silence-some-lsp-info-messages)
```
